### PR TITLE
Improvements to the Android Predictions section

### PR DESCRIPTION
**Pull request by jpignata**
*Issue #, if available:* n/a

*Description of changes:* General improvements to this section including Kotlin code samples and some general fixes for consistency.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/predictions/fragments/android/identify-entity.md [hunk 1]
```diff
@@ -123,10 +122,37 @@ public void detectEntities(Bitmap image) {
...
+
+```kotlin
+fun detectEntities(image: Bitmap) {
+    Amplify.Predictions.identify(
+        IdentifyActionType.DETECT_ENTITIES,
+        image,
+        Consumer { result: IdentifyResult ->
+            val identifyResult =
+                result as IdentifyEntityMatchesResult
+            val match = identifyResult.getEntityMatches()[0]
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/identify-entity.md:136]: Is there a `getBestMatch()` or something?

File: docs/lib/predictions/fragments/android/interpret.md [hunk 2]
```diff
@@ -2,49 +2,54 @@ The following API allows you to analyze text for language, entities (places, peo
...
-    public void interpret(String text) {
-        Amplify.Predictions.interpret(
-                text,
-                result -> Log.i("AmplifyQuickstart", result.toString()),
-                error -> Log.e("AmplifyQuickstart", error.toString(), error)
-        );
-    }
+Amplify.Predictions.interpret(
+        "I like to eat spaghetti",
+        result -> Log.i("MyAmplifyApplication", result.getSentiment().getValue().toString()),
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/interpret.md:32]: I don’t think you need the explicit `toString()`, should happen automatically no?

File: docs/lib/predictions/fragments/android/interpret.md [hunk 3]
```diff
@@ -2,49 +2,54 @@ The following API allows you to analyze text for language, entities (places, peo
...
+);
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+Amplify.Predictions.interpret(
+    "I like to eat spaghetti",
+    { result -> Log.i("MyAmplifyApplication", result.getSentiment()!!.getValue().toString()) },
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/interpret.md:43]: Need another annotation on the `getSentiment()` result, to get rid of this platform typing (`!!`)

File: docs/lib/predictions/fragments/android/translate.md [hunk 4]
```diff
@@ -1,68 +1,102 @@
...
-            result -> Log.i("AmplifyQuickstart", result.getTranslatedText()),
-            error -> Log.e("AmplifyQuickstart", error.toString())
-    );
-}
+Amplify.Predictions.translateText(
+        "I like to eat spaghetti",
+        LanguageType.ENGLISH,
+        LanguageType.RUSSIAN,
+        result -> Log.i("MyAmplifyApplication", result.getTranslatedText()),
+        error -> Log.e("AmplifMyAmplifyApplicationyQuickstart", error.toString())
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/translate.md:76]: This tag seems to have been miffed

File: docs/lib/predictions/fragments/native_common/getting-started/common.md [hunk 5]
```diff
@@ -1,39 +1,37 @@
...
+The Predictions category for enables you to integrate machine learning in your application without any prior machine learning experience.  We support a number of use cases which include, but not limited to: translating text from one language to another, converting text to speech, text recognition from an image, entities recognition, labeling real world objects, interpretation of text, and uploading images for automatic training. These use cases are powered by Amazon services including: [Amazon Translate](https://docs.aws.amazon.com/translate/latest/dg/what-is.html), [Amazon Polly](https://docs.aws.amazon.com/polly/latest/dg/what-is.html), [Amazon Transcribe](https://docs.aws.amazon.com/transcribe/latest/dg/what-is-transcribe.html), [Amazon Rekognition](https://docs.aws.amazon.com/rekognition/latest/dg/what-is.html), [Amazon Textract](https://docs.aws.amazon.com/textract/latest/dg/what-is.html), and [Amazon Comprehend](https://docs.aws.amazon.com/comprehend/latest/dg/what-is.html).
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/native_common/getting-started/common.md:1]: A rogue `for` in the first few words

Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/native_common/getting-started/common.md:1]: “Which include but are not limited to” — it sounds legal. I think we can safely cut this, and get to the point earlier

Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/native_common/getting-started/common.md:1]: Should we put the enabled use cases into an ordered or unordered list?

File: docs/lib/predictions/fragments/native_common/getting-started/common.md [hunk 6]
```diff
@@ -45,38 +43,43 @@ Enter the following when prompted:
...
 
-To push your change to the cloud, `execute the command:`
+To push your change to the cloud, **execute the command**:
+
 ```bash
 amplify push
 ```
 
-Upon completion, `awsconfiguration.json` and `amplifyconfiguration.json` should be updated to reference provisioned backend storage resources.  Note that these files should already be a part of your project if you followed the [Project setup walkthrough](~/lib/project-setup/create-application.md).
+Upon completion, `amplifyconfiguration.json` will be updated to reference provisioned backend resources.
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/native_common/getting-started/common.md:55]: To reference these newly provisioned backend resources

File: docs/lib/predictions/fragments/android/identify-entity.md [hunk 7]
```diff
@@ -48,13 +37,36 @@ public void detectEntities(Bitmap image) {
...
 
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+fun detectEntities(image: Bitmap) {
+    Amplify.Predictions.identify(
+        IdentifyActionType.DETECT_ENTITIES,
+        image,
+        Consumer { result ->
```
[the remark this answers is not in the document]
Comment (jpignata) [reply docs/lib/predictions/fragments/android/identify-entity.md:55]: I'd love a solution for that. I couldn't it to work without it, in this particular case.

File: docs/lib/predictions/fragments/android/identify-entity.md [hunk 8]
```diff
@@ -48,13 +37,36 @@ public void detectEntities(Bitmap image) {
...
+            val identifyResult = result as IdentifyEntitiesResult
+            val metadata = identifyResult.getEntities()[0]
+            Log.i("MyAmplifyApplication", metadata.getBox()!!.toShortString())
```
[the remark this answers is not in the document]
Comment (jpignata) [reply docs/lib/predictions/fragments/android/identify-entity.md:58]: Hmm, I believe it is annotated with @nullability. The `!!` is saying "this ain't null, I swear." Perhaps there's a better example here to use?

File: docs/lib/predictions/fragments/android/identify-entity.md (hunk 1 above)
Comment (jpignata) [reply docs/lib/predictions/fragments/android/identify-entity.md:136]: Negative.

File: docs/lib/predictions/fragments/android/interpret.md (hunk 2 above)
Comment (jpignata) [reply docs/lib/predictions/fragments/android/interpret.md:32]: Hmm, I don't think so. It wouldn't compile without an explicit call to `toString()` or being wrapped in `String.valueOf`:

```console
/Users/Developer/workspace/MyAmplifyApplication/app/src/main/java/com/example/myamplifyapplication/MainActivity.java:35: error: incompatible types: SentimentType cannot be converted to String
                result -> Log.i("MyAmplifyApplication", result.getSentiment().getValue()),
                                                                                      ^
```

File: docs/lib/predictions/fragments/android/interpret.md (hunk 3 above)
Comment (jpignata) [reply docs/lib/predictions/fragments/android/interpret.md:43]: Same as above, I think.

File: docs/lib/predictions/fragments/native_common/getting-started/common.md (hunk 5 above)
Comment (jpignata) [reply docs/lib/predictions/fragments/native_common/getting-started/common.md:1]: While supplies last, some restrictions apply, not available in Connecticut, you may use this plugin if under 18 with parent or guardian's permission.

Comment (jpignata) [reply docs/lib/predictions/fragments/native_common/getting-started/common.md:1]: I don't feel strongly here. I think it actually looks fine as a graf, and slightly prefer that as starting the document off with a list seems odd.

CleanShot 2020-05-23 at 15 06 12

File: docs/lib/predictions/fragments/android/identify-entity.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (drochetti) [reply docs/lib/predictions/fragments/android/identify-entity.md:55]: Tried as well and I think only Amplify KTX will be able to truly solve that

[the remark this answers is not in the document]
Comment (jpignata) [reply docs/lib/predictions/fragments/android/identify-entity.md:55]: Not great, Bob! https://developer.android.com/kotlin/interop#lambda_arguments

File: docs/lib/predictions/fragments/android/getting-started/50_translate.md [hunk 9]
```diff
@@ -1,9 +1,34 @@
...
+
 ```java
 Amplify.Predictions.translateText(
         "I like to eat spaghetti",
         LanguageType.ENGLISH,
         LanguageType.SPANISH,
-        result -> Log.i("AmplifyQuickstart", result.getTranslatedText()),
-        error -> Log.e("AmplifyQuickstart", error.toString())
+        result -> Log.i("MyAmplifyApplication", result.getTranslatedText()),
+        error -> Log.e("MyAmplifyApplication", error.toString(), error)
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/getting-started/50_translate.md:12]:
```suggestion
        error -> Log.e("MyAmplifyApplication", "Translation failed.", error)
```

File: docs/lib/predictions/fragments/android/identify-entity.md [hunk 10]
```diff
@@ -48,13 +37,36 @@ public void detectEntities(Bitmap image) {
             result -> {
                 IdentifyEntitiesResult identifyResult = (IdentifyEntitiesResult) result;
                 EntityDetails metadata = identifyResult.getEntities().get(0);
-                Log.i("AmplifyQuickstart", metadata.getBox().toShortString());
+                Log.i("MyAmplifyApplication", metadata.getBox().toShortString());
             },
-            error -> Log.e("AmplifyQuickstart", error.toString(), error)
+            error -> Log.e("MyAmplifyApplication", error.toString(), error)
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/identify-entity.md:42]:
```suggestion
            error -> Log.e("MyAmplifyApplication", "Entity detection failed.", error)
```

File: docs/lib/predictions/fragments/android/identify-entity.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jamesonwilliams) [reply docs/lib/predictions/fragments/android/identify-entity.md:58]: Whoops, I didn't understand what `!!` meant, when I wrote this. I looked it up. Sorry!

File: docs/lib/predictions/fragments/android/identify-entity.md [hunk 11]
```diff
@@ -48,13 +37,36 @@ public void detectEntities(Bitmap image) {
...
+    Amplify.Predictions.identify(
+        IdentifyActionType.DETECT_ENTITIES,
+        image,
+        Consumer { result ->
+            val identifyResult = result as IdentifyEntitiesResult
+            val metadata = identifyResult.getEntities()[0]
+            Log.i("MyAmplifyApplication", metadata.getBox()!!.toShortString())
+        },
+        Consumer { error ->
+            Log.e("MyAmplifyApplication", error.toString(), error)
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/identify-entity.md:61]:
```suggestion
            Log.e("MyAmplifyApplication", "Entity detection failed.", error)
```

File: docs/lib/predictions/fragments/android/label-image.md [hunk 12]
```diff
@@ -49,27 +38,75 @@ public void detectLabels(Bitmap image) {
...
+    Amplify.Predictions.identify(
+        LabelType.LABELS,
+        image,
+        Consumer { result: IdentifyResult ->
+            val identifyResult = result as IdentifyLabelsResult
+            val label: Label = identifyResult.getLabels()[0]
+            Log.i("MyAmplifyApplication", label.getName())
+        },
+        Consumer { error: PredictionsException ->
+            Log.e("MyAmplifyApplication", error.toString(), error)
```
Comment (jamesonwilliams) [inline docs/lib/predictions/fragments/android/label-image.md:62]:
```suggestion
            Log.e("MyAmplifyApplication", "Label detection failed.", error)
```

**Pull request merged**